api/v1alpha1/cors_types.go

```diff
@@ -29,18 +32,33 @@ type Origin string
 // CORS defines the configuration for Cross-Origin Resource Sharing (CORS).
 type CORS struct {
 	// AllowOrigins defines the origins that are allowed to make requests.
-	// +kubebuilder:validation:MinItems=1
-	AllowOrigins []Origin `json:"allowOrigins,omitempty" yaml:"allowOrigins"`
+	//
+	// +optional
+	AllowOrigins []Origin `json:"allowOrigins,omitempty"`
+
 	// AllowMethods defines the methods that are allowed to make requests.
-	// +kubebuilder:validation:MinItems=1
-	AllowMethods []string `json:"allowMethods,omitempty" yaml:"allowMethods"`
+	//
+	// +optional
```

Comment:
does unset field mean all methods are allowed ?
relates to https://github.com/envoyproxy/gateway/issues/4196

Reply:
@arkodg unset field means all methods are **not** allowed.

```yaml
  cors:
    allowOrigins:
    - http://*.foo.com
    - http://*.foo.com:80
```

```bash
➜  ~ curl -H "Origin: http://www.foo.com" \
  -H "Host: www.example.com" \
  -H "Access-Control-Request-Method: GET" \
  -X OPTIONS -v -s \
http://172.18.255.200  \
  1> /dev/null
*   Trying 172.18.255.200:80...
* Connected to 172.18.255.200 (172.18.255.200) port 80
> OPTIONS / HTTP/1.1
> Host: www.example.com
> User-Agent: curl/8.8.0-DEV
> Accept: */*
> Origin: http://www.foo.com
> Access-Control-Request-Method: GET
> 
* Request completely sent off
< HTTP/1.1 200 OK
< access-control-allow-origin: http://www.foo.com
< date: Tue, 10 Sep 2024 07:51:15 GMT
< content-length: 0
```